zebra/zebra_northbound.c

```diff
@@ -0,0 +1,2212 @@
+{
+	switch (event) {
+	case NB_EV_VALIDATE:
+	case NB_EV_PREPARE:
+	case NB_EV_ABORT:
+	case NB_EV_APPLY:
+		/* TODO: implement me. */
+		break;
+	}
+
+	return NB_ERR_NOT_FOUND;
+}
+
+/*
+ * XPath: /frr-interface:lib/interface/frr-zebra:zebra/link-detect
+ */
+static int lib_interface_zebra_link_detect_modify(enum nb_event event,
+						  const struct lyd_node *dnode,
+						  union nb_resource *resource)
+{
+	switch (event) {
+	case NB_EV_VALIDATE:
+	case NB_EV_PREPARE:
+	case NB_EV_ABORT:
+	case NB_EV_APPLY:
+		/* TODO: implement me. */
+		break;
+	}
+
+	return NB_ERR_NOT_FOUND;
+}
+
+static int lib_interface_zebra_link_detect_destroy(enum nb_event event,
+						   const struct lyd_node *dnode)
+{
+	switch (event) {
+	case NB_EV_VALIDATE:
+	case NB_EV_PREPARE:
+	case NB_EV_ABORT:
+	case NB_EV_APPLY:
+		/* TODO: implement me. */
+		break;
+	}
+
+	return NB_ERR_NOT_FOUND;
+}
+
+/*
+ * XPath: /frr-interface:lib/interface/frr-zebra:zebra/shutdown
+ */
+static int lib_interface_zebra_shutdown_modify(enum nb_event event,
+					       const struct lyd_node *dnode,
+					       union nb_resource *resource)
+{
+	switch (event) {
+	case NB_EV_VALIDATE:
+	case NB_EV_PREPARE:
+	case NB_EV_ABORT:
+	case NB_EV_APPLY:
+		/* TODO: implement me. */
+		break;
+	}
+
+	return NB_ERR_NOT_FOUND;
+}
+
+static int lib_interface_zebra_shutdown_destroy(enum nb_event event,
+						const struct lyd_node *dnode)
+{
+	switch (event) {
+	case NB_EV_VALIDATE:
+	case NB_EV_PREPARE:
+	case NB_EV_ABORT:
+	case NB_EV_APPLY:
+		/* TODO: implement me. */
+		break;
+	}
+
+	return NB_ERR_NOT_FOUND;
+}
+
+/*
+ * XPath: /frr-interface:lib/interface/frr-zebra:zebra/bandwidth
+ */
+static int lib_interface_zebra_bandwidth_modify(enum nb_event event,
+						const struct lyd_node *dnode,
+						union nb_resource *resource)
+{
+	switch (event) {
+	case NB_EV_VALIDATE:
+	case NB_EV_PREPARE:
+	case NB_EV_ABORT:
+	case NB_EV_APPLY:
+		/* TODO: implement me. */
+		break;
+	}
+
+	return NB_ERR_NOT_FOUND;
+}
+
+static int lib_interface_zebra_bandwidth_destroy(enum nb_event event,
+						 const struct lyd_node *dnode)
+{
+	switch (event) {
+	case NB_EV_VALIDATE:
+	case NB_EV_PREPARE:
+	case NB_EV_ABORT:
+	case NB_EV_APPLY:
+		/* TODO: implement me. */
+		break;
+	}
+
+	return NB_ERR_NOT_FOUND;
+}
+
+/*
+ * XPath:
+ * /frr-route-map:lib/route-map/entry/match-condition/frr-zebra:ipv4-prefix-length
+ */
+static int lib_route_map_entry_match_condition_ipv4_prefix_length_modify(
+	enum nb_event event, const struct lyd_node *dnode,
+	union nb_resource *resource)
+{
+	struct routemap_hook_context *rhc;
+	const char *length;
+	int condition, rv;
+
+	if (event != NB_EV_APPLY)
+		return NB_OK;
+
+	/* Add configuration. */
+	rhc = nb_running_get_entry(dnode, NULL, true);
+	length = yang_dnode_get_string(dnode, NULL);
+	condition = yang_dnode_get_enum(dnode, "../frr-route-map:condition");
+
+	/* Set destroy information. */
+	switch (condition) {
+	case 100: /* ipv4-prefix-length */
```

Comment:
Can we use macros here?

Reply:
Sure, is there any automatic generation? Otherwise I'd keep that for another PR (lib/routemap itself would also need this to keep consistency).

Reply:
Nothing I know of. We can write one later if not already present.